### ISSUE TITLE
using with on-prem tfs 2015 u2 server, vsoagent cannot update build details

### ISSUE THREAD

**Issue by ups216** [Microsoft/vso-agent#290]
Hi, I managed to get vsoagent connects with my on-prem tfs 2015 update 2 chs server, and the build successfully kicked off and the xcode project build successfully. However at the last stage of the process, agent reports an error in the Mac terminal saying it cannot update the build details on the tfs. 

here is the error message. 

##[Error] 2016-04-22T10:09:15.525Z: Failed Request: Forbidden(403) - 访问被拒绝: tfsadmin 对于业务流程计划 0ee9aa8e-35ad-4090-b918-5dc5cbc116fd 没有写入权限。

above that message, agent output the full json request body. I can provide the full json if someone can take a look at this. 

thanks.

**Comment by bryanmacfarlane** [comment]
Does it repro everytime?  Trying to figure out if it was a blip or if there's a permissions issue.

**Comment by ups216** [comment]
@bryanmacfarlane Yes, I can reproduce it every time. I notice there is a parameter for setting up the agent "Enter force basic (enter is false)". If I set this to false (default), the agent is not able to download the code from git repo, e.g. 

Fatal: Authentication failed for 'http://OAuth: ****** 192.168.33.34/tfs/defaultcollection/_git/test/'
/usr/bin/git failed with return code: 128

(this in case, logs are uploaded to TFS correctly). 

if I set the parameter as true, above error is gone, the repo is cloned correctly, Xocde is compiling correctly ... all tasks are running fine. But the agent is not able to upload logs to TFS.
